### ISSUE TITLE
Your thoughts on a similar project

### ISSUE THREAD

**Issue by mesudip** [jwilder/nginx-proxy#1315]
Hey there,

I am working on a project similar to this.
docker-gen is written in go which I am not that familiar with and I wanted to extend it to docker swarm, I ended up creating one from scratch but in python. 

The support for the swarm is very far from complete but it's somewhat ready for a single machine.

https://github.com/mesudip/nginx-proxy
I would like to know thoughts on it from this community.

**Comment by thomasd3** [comment]
Is it functional and ready to use?

can we do socket connections through it?

**Comment by mesudip** [comment]
It is functional and ready to use, some features are missing. I use it on production environment in our company.

Currently, it's possible to use it only as an HTTP reverse-proxy.

**Comment by thomasd3** [comment]
we have to make a decision in the next couple weeks based on the situation with this project here, so if we change, I'll definitely test yours to see if we can smoothly switch to it.

do you have any plans to support sockets?

**Comment by mesudip** [comment]
If you are referring to `TCP` and `UDP` load-balancing then Yeh, I have felt a need to add it.

I might add it after some other features are complete.

**Comment by thomasd3** [comment]
I'm talking about websockets and keeping persistent connections

**Comment by mesudip** [comment]
I am eventually planning to support the most commonly used protocols, ws/wss being one of them. 

It will take some time though.

**Comment by DaniSchenk** [comment]
@thomasd3 did you check out [traefik](https://traefik.io/)

**Comment by thomasd3** [comment]
Yes, I came from traefik to nginx-proxy; it was a while back and traefik wasn’t really working like I expected at the time and i started to look for alternatives.

Maybe it has improved in the meantime but the reality is that’s nginx-proxy works really well for us

**Comment by mesudip** [comment]
@thomasd3  

I have checked the configurations. Websocket works properly both on `jwilder/nginx-proxy` and `mesudip/nginx-proxy`. The nginx configuration allows the connection to be upgraded if requested. It is enabled by the following lines.  It's better performance wise to add it only for websocket  endpoints, but global configuration removes the hassle.

```
    map $http_upgrade $proxy_connection {
      default upgrade;
      '' close;
    }
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $proxy_connection;
```
If you want to check it yourself you can use image "mesudip/wsecho" and scale it up. Then test websocket connection with `wscat`

Can you share how are you using nginx-proxy? Are you using docker swarm or  a single machine?
How do you manage nginx-proxy when used in docker swarm with  multiple machines?

**Comment by thomasd3** [comment]
next time we deploy another machine, I will test your solution; the way we use nginx-proxy is quite simple; we have made a python script that organizes the build / push / pull / run of different modules; as one is updated, that script is called and will update the relevant docker containers on the target server. Each of these containers has tags for nginx-proxy to read so that the nginx config gets automatically updated.
